app/scripts/controllers/transactions/pending-tx-tracker.js

```diff
@@ -151,16 +155,22 @@ class PendingTransactionTracker extends EventEmitter {
       // check the network if the nonce is ahead the tx
       // and the tx has not been mined into a block
 
-      dropped = await this._checkIftxWasDropped(txMeta)
+      dropped = await this._checkIftxWasDropped(txMeta, transactionReceipt)
       // the dropped buffer is in case we ask a node for the tx
       // that is behind the node we asked for tx count
       // IS A SECURITY FOR HITTING NODES IN INFURA THAT COULD GO OUT
       // OF SYNC.
       // on the next block event it will return fire as dropped
-      if (dropped && !this.droppedBuffer[txHash]) {
-        this.droppedBuffer[txHash] = true
+      if (typeof this.droppedBuffer[txHash] !== 'number') {
+        this.droppedBuffer[txHash] = 0
+      }
+      if (dropped && this.droppedBuffer[txHash] < 3) {
         dropped = false
-      } else if (dropped && this.droppedBuffer[txHash]) {
+      }
+      ++this.droppedBuffer[txHash]
```

Comment:
Should this _always_ be incremented, even if the transaction isn't showing up as dropped?

I thought at first that this was waiting for it to be dropped on three successive checks. But currently it seems to be checking every third block instead.

Reply:
i.e. I think maybe this belongs in an `else if (dropped) { .. }` block

Reply:
Wait no, my bad, that should be `if (dropped)` not `else if (dropped)`

Reply:
hmm yeah you have a point i can move it